src/php/Compiler/Parser/ExpressionParser/AtomParser.php

```diff
@@ -80,4 +86,47 @@ public function parse(Token $token): AbstractAtomNode
 
         return new SymbolNode($word, $token->getStartLocation(), $token->getEndLocation(), Symbol::create($word));
     }
+
+    private function parseKeyword(Token $token): KeywordNode
+    {
+        $word = $token->getCode();
+        $isValid = preg_match(self::KEYWORD_REGEX, $word, $matches);
+
+        if (!$isValid) {
+            throw new KeywordParserException('This is not a valid keyword');
+        }
+
+        $isDualColon = $matches[1] === ':';
+        $hasNamespace = strlen($matches[3]) > 0;
+
+        $namespace = null;
+        if ($isDualColon && $hasNamespace) {
+            // First case is a dual colon with a namespace alias
+            // like ::foo/bar
+            $alias = $matches[3];
```

Comment:
I see that we're using `$matches[x]` by position in the regex expression. Have you consider using aliases instead of indexes/numbers? Here you can see an example of how to use aliases: https://github.com/gacela-project/gacela/blob/master/src/Framework/ClassResolver/GlobalKey.php#L18 

I am not sure if it would be possible to use them in this context, but just as an idea that worth trying because an alias is more legible than a number.